build-tools/packages/build-cli/src/library/release.ts

```diff
@@ -159,10 +173,62 @@ export function toReportKind(
 			break;
 		}
 
+		case "legacy-compat": {
+			for (const [pkg, details] of Object.entries(report)) {
+				toReturn[pkg] =
+					details.releaseGroup === "client"
+						? details.ranges.legacyCompat
+						: details.ranges.caret;
```

Comment:
This is the other place that you're checking the releaseGroup. Even though the logic here is different, I think you could unify it with one function call that will be easier to maintain and document.